# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/linux/hotfix-msm-mod-firmware.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

The msm adreno catalog is somehow missing the latest gen7 firmware mod info.

  - Rene Rebe <rene@exactcode.de>

diff --git a/drivers/gpu/drm/msm/adreno/a6xx_catalog.c b/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
index 53e2ff4406d8..935e17c385fb 100644
--- a/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
+++ b/drivers/gpu/drm/msm/adreno/a6xx_catalog.c
@@ -1490,6 +1490,18 @@ static const struct adreno_info a7xx_gpus[] = {
 };
 DECLARE_ADRENO_GPULIST(a7xx);
 
+MODULE_FIRMWARE("qcom/a702_sqe.fw");
+MODULE_FIRMWARE("qcom/a702_zap.mbn");
+MODULE_FIRMWARE("qcom/a730_sqe.fw");
+MODULE_FIRMWARE("qcom/a730_zap.mdt");
+MODULE_FIRMWARE("qcom/a740_sqe.fw");
+MODULE_FIRMWARE("qcom/gen70500_gmu.bin");
+MODULE_FIRMWARE("qcom/gen70500_sqe.fw");
+MODULE_FIRMWARE("qcom/gen70900_sqe.fw");
+MODULE_FIRMWARE("qcom/gmu_gen70000.bin");
+MODULE_FIRMWARE("qcom/gmu_gen70200.bin");
+MODULE_FIRMWARE("qcom/gmu_gen70900.bin");
+
 static inline __always_unused void __build_asserts(void)
 {
 	BUILD_BUG_ON(a630_protect.count > a630_protect.count_max);
